website/src/components/TutorialComponents/CodeBlock.tsx

```diff
@@ -165,20 +186,28 @@ function CodeEditor(props: {
                                 ))}
                             </span>}
                         <pre
+                            tabIndex={0}
                             className={clsx(
                                 _className,
                                 styles.codeBlock,
                                 'thin-scrollbar')}
                             style={{
-                                // margin: 0,
-                                // outline: "none",
                                 padding: "0",
-                                // fontFamily: "inherit",
-                                // fontSize: "inherit",
                                 ...(!props.className || !props.theme ? {} : _style),
                             }}
                             ref={editorRef}
                             spellCheck="false"
+                            onFocus={(e) => {
+                                console.log(e.target)
+                                const selectObj = window.getSelection();
+                                if (selectObj.rangeCount === 0) {
+                                    const range = new Range();
+                                    range.collapse(true);
+                                    selectObj.addRange(range);
+                                }
+                                setHasFocus(true);
+                            }}
+                            onBlur={() => setHasFocus(false)}
```

Comment:
eslint should have flagged this

Reply:
the missing `event` parameter?

Reply:
move all inlined functions to name handlers in the component body